### PR TITLE
Add /uptime command

### LINKED ISSUE #27: Add `/uptime` command to display current uptime of bot

### PR DESCRIPTION
**Pull request by koyakonsta**
fix #27 

fur science( •̀ ω •́ )✧`(*>﹏<*)′`(*>﹏<*)′(✿◡‿◡)φ(゜▽゜*)♪O(∩_∩)O(～￣▽￣)～(*^▽^*)

### CODE REVIEW

**Review by novanai — APPROVED** [review]

**Comment by JedHazaymeh** [comment]
Will review once my refactor PR is merged *(mentioned [here](https://github.com/redbrick/blockbot/pull/32#issuecomment-1921144129) too)*

File: src/extensions/uptime.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+from datetime import datetime
+
+import arc
+
+start_time = datetime.now()
+
+plugin = arc.GatewayPlugin("Blockbot Uptime")
+
+
```
Comment (JedHazaymeh) [inline src/extensions/uptime.py:9]: **nit** I would suggest only using single newlines:
```suggestion
```

Comment (koyakonsta) [reply src/extensions/uptime.py]: ¯\(°_o)/¯

File: src/extensions/uptime.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+plugin = arc.GatewayPlugin("Blockbot Uptime")
+
+
+@plugin.include
+@arc.slash_command("uptime", "Show formatted uptime of Blockbot")
+async def uptime(ctx):
+    up_time = datetime.now() - start_time
+    h, ms = divmod(up_time.seconds, 3600)
+    m, s = divmod(ms, 60)
+    await ctx.respond(f"Uptime: **{up_time.days} days, {h} hours, {m} minutes, and {s} seconds**.")
```
Comment (JedHazaymeh) [inline src/extensions/uptime.py:15]: If `h == 1`, would it be displayed as `1 hours` in the output? If so, do we want to correct this to `1 hour`?

Comment (JedHazaymeh) [reply src/extensions/uptime.py:15]: Same applies to the other values.

Comment (koyakonsta) [reply src/extensions/uptime.py:15]: something like this?


```
d = up_time.days
plurals = tuple('s' * (i!=1) for i in [d,h,m,s])
await ctx.respond(f"Uptime: **{d} day%s, {h} hour%s, {m} minute%s, and {s} second%s**." % plurals)
```

File: src/extensions/uptime.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+plugin = arc.GatewayPlugin("Blockbot Uptime")
+
+@plugin.include
+@arc.slash_command("uptime", "Show formatted uptime of Blockbot")
+async def uptime(ctx):
+    up_time = datetime.now() - start_time
+    h, ms = divmod(up_time.seconds, 3600)
+    m, s = divmod(ms, 60)
+    await ctx.respond(f"Uptime: **{up_time.days} days, {h} hours, {m} minutes, and {s} seconds**.")
```
Comment (JedHazaymeh) [inline src/extensions/uptime.py:15]: I did something like this, which handles plurals and omits 0 values:
```suggestion
    d = up_time.days
    h, ms = divmod(up_time.seconds, 3600)
    m, s = divmod(ms, 60)

    format = lambda val, str: f"{val} {str}{'s' if val != 1 else ''}"
    message_parts = [(d, "day"), (h, "hour"), (m, "minute"), (s, "second")]
    formatted_parts = [format(val, str) for val, str in message_parts if val]
    
    await ctx.respond(f"Uptime: **{', '.join(formatted_parts)}**")
```
image

**Review by JedHazaymeh — CHANGES_REQUESTED** [review]

File: src/extensions/uptime.py (hunk 3 above)
Comment (koyakonsta) [reply src/extensions/uptime.py]: it's a bit longer _but_  the timestamp's more naturally worded - i like it

Comment (JedHazaymeh) [reply src/extensions/uptime.py]: Longer isn't an issue in most cases, just readability by design! :+1:

**Review by JedHazaymeh — APPROVED** [review]

**Pull request merged**